app/src/main/java/com/example/next_show/data/RecommendationClient.java

```diff
@@ -38,59 +40,48 @@ public RecommendationClient(Shows showsObj) {
         this.showsObj = showsObj;
     }
 
-    public void fetchRecommendedShows(List<String> savedShows, FeedFragment.ShowCallback showCallback, FeedFragment.GenreMatchedCallback gmCallback) {
-        this.genreMatchedCallback = gmCallback;
-        this.showCallback = showCallback;
-
-        if (savedShows.isEmpty()){
-            // if no shows to compare, call user genre matched ones
-            fetchGenreMatchedShows(showsObj, genreMatchedCallback);
+    public void fetchRelatedShows(List<String> savedShows, ResponseCallback callback) {
+        // check if no saved shows and immediately fail
+        if(savedShows.isEmpty() || savedShows == null) {
+            callback.onFailure(NO_CALL);
```

Comment:
You should check null first.  What will happen if the savedShows is `null` but you first try to check if it's empty?

I think you should name NO_CALL something a bit more descriptive like ERROR_NO_SAVED_SHOWS_TO MATCH_RELATED